altos-rust/port/cortex-m0/src/peripheral/usart/control.rs

```diff
@@ -201,36 +442,69 @@ impl CR1 {
 /// CR2
 // ------------------------------------
 
-/// There are four stop bit settings: .5, 1, 1.5, 2
-#[derive(Copy, Clone)]
+/// Defines the possible StopLength configurations for the Usart.
+/* Half: 0.5 stop bit
+ * One: 1 stop bit
+ * OneAndHalf: 1.5 stop bits
+ * Two: 2 stop bits
+ */
+#[derive(Copy, Clone, Debug)]
 pub enum StopLength {
     Half,
     One,
     OneAndHalf,
     Two,
 }
 
-#[derive(Copy, Clone)]
+/// Defines the base address variable for control register two.
+#[derive(Copy, Clone, Debug)]
 struct CR2 {
     base_addr: *const u32,
 }
 
+/// Implements the Register trait for Usartx_CR2.
+/// Stores base address for control register two, which is the address
+/// of the Usart being used to access this register. Uses the base address
+/// combined with the register offset to calculate the register address.
```

Comment:
Reference page 1753 of the Rust documentation guide for people who can't comment good and want to do other stuff good too